dist/PreserveModuleNamePlugin.js

```diff
@@ -73,6 +73,29 @@ function getPreservedModules(modules) {
                 m[exports.preserveModuleName] = req;
                 return true;
             }
+            // Since its relative, link it with the module alias if possible
+            const reasonRequest = r.module && r.module.rawRequest;
+            const moduleDependency = r.dependency && r.dependency.module;
+            const moduleRequest = moduleDependency && removeLoaders(moduleDependency.request);
+            if (req && aliases && reasonRequest && moduleRequest) {
+                const aliasMatches = Object.keys(aliases).filter((alias) => {
+                    const aliasPath = aliases[alias];
+                    const matchPath = !!(moduleRequest.match(new RegExp(`^${aliasPath}`)));
+                    const matchKey = !!(reasonRequest.match(new RegExp(`^${alias}`)));
+                    // We want to match against the alias file path AND the path that is being requested
+                    // This allows us to use relative files inside aliased files
+                    return (matchPath && matchKey);
+                });
+                // Invalid?
+                if (aliasMatches.length > 1) {
+                    throw new Error(`Incorrect alias usage. "${reasonRequest}" is duplicated in ${aliasMatches}`);
+                }
+                else if (aliasMatches.length === 1) {
+                    const aliasKey = aliasMatches[0];
+                    m[exports.preserveModuleName] = moduleRequest.replace(aliases[aliasKey], aliasKey);
```

Comment:
That has to be wrong. You set the module id to a name that comes from another module, modulo aliases resolution?!
If it's correct there's something I don't understand here.